cli/src/utils/nice_error.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+/*
+A nice error type that allows you to associate a longer description, a
+source file and suggestions with it.
+*/
+
+const chalk = require('chalk');
+
+class NiceError extends Error {
+  constructor(message, options) {
+    super(message);
+    const opts = options || {};
+    this.description = opts.description || null;
+    this.suggestions = opts.suggestions || null;
+    // TODO: maybe allow multiple source locations and spans of text
+    // instead of a single column offset
+    this.src = (opts.src) ? Object.assign({}, opts.src) : null;
+  }
+  toString() {
+    return this.message;
+  }
+
+  niceString(options) {
+    const opts = options || {};
+    const cSize = opts.contextSize != null ? opts.contextSize : 2;
+    const results = [ this.message ];
+    if (this.description) {
+      results.push('', this.description);
+    }
+    if (this.src != null) {
+      const formattedSrc = NiceError._formatContext(
+        this.src.contents,
+        this.src.line,
+        this.src.column,
+        cSize
+      );
+      if (formattedSrc.length > 0) {
+        results.push(`\nIn ${this.src.filename}, ` +
+                     `line ${this.src.line}, ` +
+                     `column ${this.src.column}:`);
+        results.push.apply(results, formattedSrc);
+      }
+    }
+    if (this.suggestions) {
+      results.push(
+        '', // extra newline before suggestions
+        chalk.red(
+          this.suggestions.length > 1 ? 'Suggestions:' : 'Suggestion:'));
+      results.push.apply(
+        results, this.suggestions.map((note) => `  ➤ ${note}`));
```

Comment:
We use unicode elsewhere, and I tried using *, the problem was with very long suggestions it wasn't visually obvious where one started and the next began